src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -104,33 +106,17 @@ public void Constructor_TimerIsInitialized()
             VerifyTimerStart(Times.Once());
             timerRunning.Should().Be(true);
 
-            // The call to service.IsConnected is on a background thread so we can't
-            // reliably check whether it has occurred yet. However, we know that 
-            // service.GetIssues(...) should never have been called since the service
-            // is not connected.
-            VerifyServiceGetIssues(Times.Never());
+            WaitForInitialFetchTaskToStart();
+            VerifyServiceIsConnected(Times.Exactly(2));
+            VerifyServiceGetIssues(Times.Once());
 
             // 2. Timer event raised -> check attempt is made to synchronize data
             RaiseTimerElapsed(DateTime.UtcNow);
 
-            VerifyServiceIsConnected(Times.Exactly(2));
-            VerifyServiceGetIssues(Times.Never()); // still not connected so can't get data
+            VerifyServiceIsConnected(Times.Exactly(3));
+            VerifyServiceGetIssues(Times.Exactly(2));
             timerRunning.Should().Be(true);
-        }
 
-        [TestMethod]
-        public void Constructor_Disconnected_StartsMonitoring_SkipsSyncing()
```

Comment:
This is no longer the behaviour we want - the background task will now wait for the connection.
The new behaviour is covered by a later test.